lib/main.js

```diff
@@ -66,21 +66,40 @@ function run(actionInput) {
                     return 2; // Exit the function early
                 }
                 const should_fail = core.getInput('fail_on_error');
+                const should_fail_on_level = core.getInput('fail_level');
                 // Pipe to reviewdog ...
                 core.info('Calling reviewdog 🐶');
                 process.env['REVIEWDOG_GITHUB_API_TOKEN'] = core.getInput('token');
-                return yield exec.exec(actionInput.reviewdogPath, [
-                    '-f=rdjsonl',
-                    `-name=vale`,
-                    `-reporter=${core.getInput('reporter')}`,
-                    `-fail-on-error=${should_fail}`,
-                    `-filter-mode=${core.getInput('filter_mode')}`,
-                    `-level=${vale_code == 1 && should_fail === 'true' ? 'error' : 'info'}`
-                ], {
-                    cwd,
-                    input: Buffer.from(output.stdout, 'utf-8'),
-                    ignoreReturnCode: true
-                });
+                // Support deprecated -fail-on-error option 
+                if (should_fail === 'true' && should_fail_on_level === 'none') {
+                    return yield exec.exec(actionInput.reviewdogPath, [
+                        '-f=rdjsonl',
+                        `-name=vale`,
+                        `-reporter=${core.getInput('reporter')}`,
+                        `-fail-on-error=true`,
+                        `-filter-mode=${core.getInput('filter_mode')}`,
+                        `-level=${vale_code == 1 ? 'error' : 'info'}`
+                    ], {
+                        cwd,
+                        input: Buffer.from(output.stdout, 'utf-8'),
+                        ignoreReturnCode: true
+                    });
+                } else {
+                    return yield exec.exec(actionInput.reviewdogPath, [
+                        '-f=rdjsonl',
+                        `-name=vale`,
+                        `-reporter=${core.getInput('reporter')}`,
+                        `-fail-level=${should_fail_on_level}`,
+                        `-filter-mode=${core.getInput('filter_mode')}`,
+                        `-level=${vale_code == 1 && (should_fail_on_level === 'error' ||
+                                                     should_fail_on_level === 'warning' ||
+                                                     should_fail_on_level === 'info') ? should_fail_on_level : 'info'}`
```

Comment:
```suggestion
                        `-level=${vale_code == 1 && (should_fail_on_level === 'error' ||
                                                     should_fail_on_level === 'warning' ||
                                                     should_fail_on_level === 'info') ? should_fail_on_level : 'info'}`
```
I'm not sure there is a need to check `should_fail_on_level` value. If that's the case, you might need to add `'any'` to be aligned with expected values. I don't think that other cases should be handled (`'default'` and `'none'`) since these cases don't make reviewdog exit with 1 (see https://github.com/reviewdog/reviewdog/blob/c53f10c90a146a361c50fe148a78970873ba40ad/fail_level.go#L67)

Reply:
After reviewing what I wrote, I agree with your comments. I'll remove the check.

In the longer run, I imagine that the goal is to be able to define `-level` directly.